### PR TITLE
Modigied the Gemfile : installed Devise gem and Devise Invitable , ad…

### PR DESCRIPTION
**Pull request by urvashivasita23**
…ded client side validation in JS file.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem 'devise_invitable'
 gem 'kaminari'
 gem 'ransack'
 gem 'webpacker', '~> 5.0'
+# gem "jquery-validation-rails"
```
Comment (dipak-tntra) [inline Gemfile:19]: remove commented unwanted gems.

File: app/assets/stylesheets/admin/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the admin/Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (dipak-tntra) [inline app/assets/stylesheets/admin/users.scss:3]: remove this unwanted file if you are not using it.

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# user module
+module Admin
+  class UsersController < ApplicationController
+    def index
+    end
+    # def check_email
+    #   @user = User.find_by_email(params[:user][:email])
+    #   respond_to do |format|
+    #    format.json { render :json => !@user }
+    #   end
```
Comment (dipak-tntra) [inline app/controllers/admin/users_controller.rb:12]: remove unwanted commented code.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,2 +1,5 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
```
Comment (dipak-tntra) [inline app/controllers/application_controller.rb:3]: add class level comments.

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# user module
```
Comment (JigarShahTntra) [inline app/controllers/admin/users_controller.rb:3]: give appropriate comments. Please!!!!

File: app/models/user.rb [hunk 6]
```diff
@@ -1,3 +1,12 @@
+# frozen_string_literal: true
+
 class User < ApplicationRecord
```
Comment (JigarShahTntra) [inline app/models/user.rb:3]: Style/Documentation: Missing top-level class documentation comment.

File: app/controllers/admin/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+# Admin-module
+module Admin
+  # user-controller
```
Comment (JigarShahTntra) [inline app/controllers/admin/users_controller.rb:5]: no capitalize?

File: app/controllers/application_controller.rb [hunk 8]
```diff
@@ -1,2 +1,6 @@
+# frozen_string_literal: true
+
+# application-controller
```
Comment (JigarShahTntra) [inline app/controllers/application_controller.rb:3]: capitalize the comment.

File: app/javascript/application.js [hunk 9]
```diff
@@ -0,0 +1,3 @@
+//= require jquery.min.js
+//= require jquery-validate.js
+//= require login.js
```
Comment (JigarShahTntra) [inline app/javascript/application.js:3]: add empty line at the end of the document.

File: app/javascript/login.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+            messages: {
+                'user[email]': {
+                  required: "The Email is required!",
+                  email: "Please enter a valid email address!"
+                }
+            }
+        });
+      }      
+    };
+
```
Comment (JigarShahTntra) [inline app/javascript/login.js:27]: add new line at the end of the document.

File: app/javascript/packs/application.js [hunk 11]
```diff
@@ -3,13 +3,16 @@
...
 Rails.start()
 Turbolinks.start()
 ActiveStorage.start()
+require("channels");
+require("jquery");
+import "bootstrap";
+import "../stylesheets/application";
+//require("packs/jquery-validate");
+
+
```
Comment (JigarShahTntra) [inline app/javascript/packs/application.js:18]: move import and require to app/assets/javacripts/application.js

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -7,10 +7,14 @@
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+
   </head>
```
Comment (JigarShahTntra) [inline app/views/layouts/application.html.erb:13]: remove empty lines from the html document.

File: app/views/users/confirmations/new.html.erb [hunk 13]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
```
Comment (JigarShahTntra) [inline app/views/users/confirmations/new.html.erb:8]: move logical part to the helper.

resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email

File: app/views/users/confirmations/new.html.erb [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
```
Comment (JigarShahTntra) [inline app/views/users/confirmations/new.html.erb:5]: remove empty lines from the html document.

File: app/views/users/mailer/email_changed.html.erb [hunk 15]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
+
+<% if @resource.try(:unconfirmed_email?) %>
+  <p>We're contacting you to notify you that your email is being changed to <%= @resource.unconfirmed_email %>.</p>
+<% else %>
+  <p>We're contacting you to notify you that your email has been changed to <%= @resource.email %>.</p>
+<% end %>
```
Comment (JigarShahTntra) [inline app/views/users/mailer/email_changed.html.erb:7]: move to one line and use helper remove the if else.

File: app/views/users/registrations/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+# #     messages: {
+# #       "user[email]": {
+# #         required: "Please, enter a reference",
+# #         minlength: 'Please enter exact 6 numbers only',
+# #         email: 'Please include @'
+# #       }
+# #     }
+# #   });
+# # });
+# # </script>
```
Comment (JigarShahTntra) [inline app/views/users/registrations/new.html.erb:58]: remove commented code.

**Pull request merged**